lib/polyfill/patchedmediakeys_ms.js

```diff
@@ -519,84 +518,16 @@ shaka.polyfill.PatchedMediaKeysMs.MediaKeySession.prototype.
 shaka.polyfill.PatchedMediaKeysMs.onMsNeedKey_ = function(event) {
   shaka.log.debug('PatchedMediaKeysMs.onMsNeedKey_', event);
 
-  // Alias
-  const PatchedMediaKeysMs = shaka.polyfill.PatchedMediaKeysMs;
-
   // NOTE: Because "this" is a real EventTarget, on IE, the event we dispatch
   // here must also be a real Event.
   let event2 = /** @type {!CustomEvent} */(document.createEvent('CustomEvent'));
   event2.initCustomEvent('encrypted', false, false, null);
   event2.initDataType = 'cenc';
-  event2.initData = PatchedMediaKeysMs.normaliseInitData_(event.initData);
+  event2.initData = shaka.util.Pssh.normaliseInitData(event.initData);
 
   this.dispatchEvent(event2);
 };
 
-
-/**
- * Normalise the initData array. This is to apply browser specific work-arounds,
- * e.g. removing duplicates which appears to occur intermittently when the
- * native msneedkey event fires (i.e. event.initData contains dupes).
- *
- * @param {?Uint8Array} initData
- * @private
- * @return {?Uint8Array}
- */
-shaka.polyfill.PatchedMediaKeysMs.normaliseInitData_ = function(initData) {
```

Comment:
Why are you moving something out of the polyfills? This can have dangerous side effects. If you just needed the share code, it would be safer to have the polyfill call the common code than to remove the polyfill.

Reply:
This just appears to be a utility method, so it should be fine to move it out of here.

Reply:
the function we moved is indeed a utility function to normalise the PSSH data. This was originally written by the shaka team for the MS polyfill, but we also require it for the parsing of the segments so we moved it out.